core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpSamDAO.scala

```diff
@@ -310,3 +378,11 @@ class HttpSamDAO(baseSamServiceURL: String, serviceAccountCreds: Credential)(imp
     }
   }
 }
+
+class SpanSettingInterceptor(span: Span) extends Interceptor {
+  override def intercept(chain: Interceptor.Chain): Response = {
+    Using(Tracing.getTracer.withSpan(span)) { _ =>
+      chain.proceed(chain.request())
```

Comment:
The tracing library we use was built for java and passed the trace information through thread local variables. That does not work in scala using `Futures` or other delayed execution since we are switching threads all the time. So this interceptor contains the tracing span to propagate and calls `withSpan` when we have gotten to the right thread.